api/service.go

```diff
@@ -536,19 +692,19 @@ func (s *service) AddFile(server pb.API_AddFileServer) error {
 	if err != nil {
 		return err
 	}
-	var name, projID string
-	switch payload := req.GetPayload().(type) {
+	var filePath string
+	switch payload := req.Payload.(type) {
 	case *pb.AddFileRequest_Header_:
-		projID = payload.Header.ProjectID
-		name = payload.Header.Name
+		filePath = payload.Header.Path
 	default:
-		return fmt.Errorf("project ID is required")
+		return fmt.Errorf("add file header is required")
 	}
-
-	proj, err := s.getProjectForScope(server.Context(), projID, scope)
+	folderName, fileName := parsePath(filePath)
+	folder, err := s.getFolderWithScope(server.Context(), folderName, scope)
```

Comment:
Here we have the folder name from the path and can see if the current user session is authorized to manage this folder.